src/postgres/commands/PostgresQueryWizard/FunctionQuerySteps/FunctionQueryReturnTypeStep.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { QuickPickItem } from "vscode";
+import { AzureWizardPromptStep, IWizardOptions } from "vscode-azureextensionui";
+import { ext } from "../../../../extensionVariables";
+import { localize } from "../../../../utils/localize";
+import { FunctionQueryCustomReturnTypeStep } from "./FunctionQueryCustomReturnTypeStep";
+import { IPostgresFunctionQueryWizardContext } from "./IPostgresFunctionQueryWizardContext";
+
+const customReturnTypePick: QuickPickItem = { label: localize('enterCustomReturnType', '$(pencil) Enter custom return type...') };
+
+export class FunctionQueryReturnTypeStep extends AzureWizardPromptStep<IPostgresFunctionQueryWizardContext> {
+    public async prompt(wizardContext: IPostgresFunctionQueryWizardContext): Promise<void> {
+        const returnTypeQuickPicks: QuickPickItem[] = returnTypes.map(r => { return { label: r }; });
+        returnTypeQuickPicks.push(customReturnTypePick);
+
+        wizardContext.returnTypePick = (await ext.ui.showQuickPick(returnTypeQuickPicks, {
+            placeHolder: localize('selectReturnType', 'Select return type')
+        }));
+    }
+
+    public shouldPrompt(): boolean {
+        return true;
```

Comment:
In general, `shouldPrompt` should be relative to the thing you're prompting about. So in this case:
```
public shouldPrompt(wizardContext: IPostgresFunctionQueryWizardContext): boolean {
    return !wizardContext.returnType;
}
```

The idea is that if someone sets the returnType before the wizard starts (aka some sort of smart default), it won't prompt for this step.